xbmc/network/websocket/WebSocket.cpp

```diff
@@ -104,7 +108,7 @@ CWebSocketFrame::CWebSocketFrame(const char* data, uint64_t length)
   // Get the mask
   if (m_masked)
   {
-    m_mask = *(const uint32_t *)(m_data + LENGTH_MIN + offset);
+    std::memcpy(&m_mask, m_data + LENGTH_MIN + offset, 4);
```

Comment:
```suggestion
    std::memcpy(&m_mask, m_data + LENGTH_MIN + offset, offset += 4);
```
though hard to read 🤪

Reply:
I think this is wrong. Order of evaluation of function arguments is unspecified, so offset used in third argument may be the initial `offset`, or `offset+4` - whichever the compiler prefers. See [here](https://stackoverflow.com/questions/38501587/what-are-the-evaluation-order-guarantees-introduced-by-c17).

Reply:
indeed, a pitfall i‘ve read about some time ago and just forgot 😊 thx popcornmix